lib/SILOptimizer/Transforms/TempRValueElimination.cpp

```diff
@@ -268,9 +269,22 @@ collectLoads(Operand *addressUse, CopyAddrInst *originalCopy,
     loadInsts.insert(user);
     return true;
   }
-  case SILInstructionKind::LoadBorrowInst:
+  case SILInstructionKind::LoadBorrowInst: {
     loadInsts.insert(user);
-    return true;
+    BorrowedValue borrow(cast<LoadBorrowInst>(user));
+    auto visitEndScope = [&](Operand *op) -> bool {
+      auto *opUser = op->getUser();
+      if (auto *endBorrow = dyn_cast<EndBorrowInst>(opUser)) {
```

Comment:
You also need to check if the end_borrow is in `block`.

Reply:
right, I'll fix it.